### ISSUE TITLE
[Q] Current time in transiet prompt before command execution

### ISSUE THREAD

**Issue by lvsv** [romkatv/powerlevel10k#917]
Hi there,

I want to add current time before command execution, right it was noticed at https://github.com/romkatv/powerlevel10k/issues/564#issuecomment-599057666

> "I believe you already know how to configure transient prompt this way. Let me know if not."

@romkatv, can you explain how to get it?

Thanks!

**Comment by romkatv** [comment]
```zsh
POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(dir newline prompt_char)
POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=(time newline time)
POWERLEVEL9K_PROMPT_CHAR_BACKGROUND=
POWERLEVEL9K_PROMPT_CHAR_LEFT_PROMPT_LAST_SEGMENT_END_SYMBOL=
POWERLEVEL9K_PROMPT_CHAR_LEFT_PROMPT_FIRST_SEGMENT_START_SYMBOL=
POWERLEVEL9K_PROMPT_CHAR_LEFT_LEFT_WHITESPACE=
POWERLEVEL9K_PROMPT_CHAR_LEFT_RIGHT_WHITESPACE=
POWERLEVEL9K_TIME_UPDATE_ON_COMMAND=true
p10k-on-post-prompt() { p10k display '1'=hide '2/right/time'=show }
p10k-on-pre-prompt()  { p10k display '1'=show '2/right/time'=hide }
```

**Comment by billywitherspoon** [comment]
I stumbled across this trying to add right timestamps to transient prompts. I was successful by following the above configurations AND setting `typeset -g POWERLEVEL9K_TRANSIENT_PROMPT=off`.

The custom code effectively turns the prompt to transient mode and if `POWERLEVEL9K_TRANSIENT_PROMPT` is set to `always` it will conflict with properly showing the timestamp on previous commands.

**Comment by billywitherspoon** [comment]
Yup. Just wanted to be explicit for those such as myself that were attempting to implement this feature after enabling transient prompt.

**Comment by brandon-fryslie** [comment]
I would like to thank @romkatv for leaving such amazing information in these GitHub issues on how to do so many things with powerline10k.  I have a capable custom prompt I cobbled together over the years that displays a lot of relevant information in a very readable way.  However I wanted to try something new and to leverage some existing tooling rather than do everything from scratch. 

After several disappointing attempts over the years with starship and spaceship, where I ended up with MORE code for LESS functionality just to configure them than I had in my custom prompt in the first place (and with far less functionality), I gave up on them.  The defaults are almost useless and then you're locked into a system where customization is more difficult than having a simple entirely custom prompt.  

Then I decided to give powerline10k a shot.  Within hours I had matched the functionality of my previous prompt, it's a lot more beautiful, I can modify it to do anything I want quite easily, and there are so many things it can do that I had never thought of and even if I had, would never have had time to implement.  With the information in these GitHub issues, there is enough information for someone with basic (or maybe intermediate) zsh skills to get what they want in a reasonable amount of time and effort.  

Thanks again for all your efforts, it will continue to pay huge dividends to others like me who can stand on the shoulders of giants and achieve amazing results.  Looking to try out zsh4humans sometime in the next 5 years and be surprised and delighted yet again.  Sometimes it's convenient for an ecosystem to move at a leisurely pace.  :)

**Comment by romkatv** [comment]
@brandon-fryslie I'm sorry about what you had to go through. I know you know what I mean :D. In my defense, it was my starter project in zsh :P

Seriously though, I'm thankful to fate (or whoever is in charge of things that happen) that I made p10k. I felt guilty much of the time I worked on it (I thought I was supposed to spend my time elsewhere), but seeing genuine positive feedback like yours corrects the past: I was doing what was good and right, all the while being mistaken about my duties. Thank you for your message--it means a lot!
